codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/localservice/DafnyLocalServiceGenerator.java

```diff
@@ -303,8 +338,29 @@ func NewClient(clientConfig $L) (*$T, error) {
                     symbolProvider.toSymbol(outputShape)
                   )
                 );
+            returnError = "return nil,";
           }
-          returnError = "return nil,";
+        }
+        String validationCheck = "";
+        // inputShape is not a structure only when we have a positional trait
+        // TODO: Figure out how to validate constraint in a structure with positional trait
```

Comment:
This is a general Smithy-Dafny gap... I think... 
I hope that we deprecate positional.

It goes against Smithy's philosophy and is PITA to implement.